host/src/containers/GameSessionContainer.tsx

```diff
@@ -184,40 +184,39 @@ const GameSessionContainer = () => {
         const teamDataRequests = response.teams.map(team => {
           return apiClient.getTeam(team.id); // got to call the get the teams from the API so we can see the answers
         });
-    
-        Promise.all(teamDataRequests) 
+
+        Promise.all(teamDataRequests)
           .then(responses => {
-            setTeamsArray(responses); 
-            setIsModalOpen(false); 
+            setTeamsArray(responses);
+            setIsModalOpen(false);
           })
           .catch(reason => console.log(reason));
       });
-    
+
   };
 
   const handleStartGame = () => {
     // I'm keeping this console.log in until we figure out NOT_STARTED so we can tell there's been a change in state 
-    console.log(gameSession.currentState);  
-    if (gameSession.currentState === GameSessionState.TEAMS_JOINING)
-    {
+    console.log(gameSession.currentState);
+    if (gameSession.currentState === GameSessionState.TEAMS_JOINING) {
+      handleUpdateGameSession({ currentState: GameSessionState.CHOOSE_CORRECT_ANSWER, currentQuestionIndex: 0 });
       setIsTimerActive(true);
-      setIsModalOpen(true);   
+      setIsModalOpen(true);
     }
-    else
-      handleUpdateGameSession({currentState: GameSessionState.TEAMS_JOINING, currentQuestionIndex: 0});
   };
 
   if (!gameSession) {
     return null;
   };
 
-
   switch (gameSession.currentState) {
     case GameSessionState.NOT_STARTED:
     case GameSessionState.TEAMS_JOINING:
-      return <StartGame {...gameSession} gameSessionId={gameSession.id} isTimerActive={isTimerActive} isModalOpen={isModalOpen} handleStartGameModalTimerFinished={handleStartGameModalTimerFinished} handleStartGame={handleStartGame}/>;
-
+      return <StartGame {...gameSession} gameSessionId={gameSession.id} isTimerActive={isTimerActive} isModalOpen={isModalOpen} handleStartGameModalTimerFinished={handleStartGameModalTimerFinished} handleStartGame={handleStartGame} />;
     case GameSessionState.CHOOSE_CORRECT_ANSWER:
+      if (isModalOpen) {
+        return <StartGame {...gameSession} gameSessionId={gameSession.id} isTimerActive={isTimerActive} isModalOpen={isModalOpen} handleStartGameModalTimerFinished={handleStartGameModalTimerFinished} handleStartGame={handleStartGame} />;
+      }
```

Comment:
Is there a case that `isModalOpen` is `false`? Is it intentional that if that's `false,` it should continue to other cases below?

Reply:
I believe isModalOpen is false when the countdown from 3 is at 0. I wasn't quite sure how to keep the functionality of the timer without the if statement wrapping around the `return <StartGame {...gameSession} gameSessionId={gameSession.id} isTimerActive={isTimerActive} isModalOpen={isModalOpen} handleStartGameModalTimerFinished={handleStartGameModalTimerFinished} handleStartGame={handleStartGame} />;` since without the if statement, the start screen would re-appear as soon as the countdown finished. Since isModalOpen is already being passed in to `<StartGame {...gameSession} gameSessionId={gameSession.id} isTimerActive={isTimerActive} isModalOpen={isModalOpen} handleStartGameModalTimerFinished={handleStartGameModalTimerFinished} handleStartGame={handleStartGame} />;`, I also tried simply excluding this line from the CHOOSE_CORRECT_ANSWER case of the switch statement and have the previous case from TEAMS_JOINING bleed over but this would result in the countdown disappearing entirely immediately after appearing. I'll continue to look for other approaches to this.